### PR TITLE
Switch from distutils to setuptools

### PR DESCRIPTION
**Pull request by jelly**
distutils is deprecated upstream.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,63 +12,44 @@
...
+                          'src/options.h',
+                          'src/package.h',
+                          'src/pyalpm.h',
+                          'src/util.h'])
 
 
 with open("README", "r") as fh:
     long_description = fh.read()
 
+pycman_scripts = ['database', 'deptest', 'query', 'remove', 'sync', 'upgrade', 'version']
```
Comment (SantiagoTorres) [inline setup.py]: I wonder if we'd like to name this "variable" in uppercase, so as to make it clear it's a "constant"

File: .travis.yml [hunk 2]
```diff
@@ -8,13 +8,13 @@ arch:
 
   packages:
     - python
-    - python-nose
+    - python-pytest
```
Comment (SantiagoTorres) [inline .travis.yml]: I wonder if we'd like to have any dependency here at all. Can't we use `python -m unittest` or use a very small test aggregator script?

I'm also down for moving to pytest later down the line if that's the plan...

File: setup.py [hunk 3]
```diff
@@ -13,63 +12,44 @@
...
-      cmdclass={
-          'test': TestCommand
-      },
       classifiers=[
-        'Development Status :: 6 - Mature',
-        'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
-        'Programming Language :: C',
+          'Development Status :: 6 - Mature',
+          'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
+          'Programming Language :: C',
```
Comment (SantiagoTorres) [inline setup.py]: I also wonder if we want to drop:
```
'Topic :: System :: Software Distribution '
'Topic :: System :: Systems Administration'
```

In there.

File: .travis.yml (hunk 2 above)
Comment (jelly) [reply .travis.yml]: My other PR makes extensive use of pytest for coverage and pytest allows filtering and dropping into a pdb shell. It has so many advantages.

File: setup.py (hunk 1 above)
Comment (jelly) [reply setup.py]: seems alright :)

File: setup.py (hunk 3 above)
Comment (jelly) [reply setup.py]: sounds good

File: .travis.yml (hunk 2 above)
Comment (SantiagoTorres) [reply .travis.yml]: Ah yeah, so disregard this please :+1:

**Comment by eli-schwartz** [comment]
You can run pytest without specifying "tests", one of the cool things about using high-level wrappers like pytest rather than low-level unittest is that it does intelligent autodiscovery of the test suite no matter where it is located.

**Pull request merged**